modules/siddhi-query-api/src/main/java/io/siddhi/query/api/util/AnnotationHelper.java

```diff
@@ -22,14 +22,23 @@
 import io.siddhi.query.api.annotation.Element;
 import io.siddhi.query.api.exception.DuplicateAnnotationException;
 
+import java.util.Arrays;
 import java.util.List;
 
 /**
  * Helper class to extract annotations
  */
 public class AnnotationHelper {
 
-    // TODO: 1/28/17 update helper methods to work with nested annotations.
+    public static Annotation getNestedAnnotation(List<Annotation> annotationList, String... annotationNames) {
```

Comment:
Why is this ?

Reply:
to get nested annotations like cache annotation inside store annotation. usage - https://github.com/niruhan/siddhi/blob/53466a751c2bea850aa9fed867fd9e61d903f2c9/modules/siddhi-core/src/main/java/io/siddhi/core/table/record/AbstractQueryableRecordTable.java#L107. Can go to any depth of nesting recursively we just need to give the annotation names in order to traverse as a string[]